src/Material/Menu.elm

```diff
@@ -0,0 +1,523 @@
+module Material.Menu
+  ( Model, model
+  , Item, item
+  , Action, update
+  , topLeft
+  , bottomLeft
+  , topRight
+  , bottomRight
+  , unaligned
+  , view
+  ) where
+
+{-| From the [Material Design Lite documentation](http://www.getmdl.io/components/#menus-section):
+
+> The Material Design Lite (MDL) menu component is a user interface element
+> that allows users to select one of a number of options. The selection
+> typically results in an action initiation, a setting change, or other
+> observable effect. Menu options are always presented in sets of two or
+> more, and options may be programmatically enabled or disabled as required.
+> The menu appears when the user is asked to choose among a series of
+> options, and is usually dismissed after the choice is made.
+
+> Menus are an established but non-standardized feature in user interfaces,
+> and allow users to make choices that direct the activity, progress, or
+> characteristics of software. Their design and use is an important factor in
+> the overall user experience. See the menu component's Material Design
+> specifications page for details.
+
+See also the
+[Material Design Specification]([https://www.google.com/design/spec/components/menus.html).
+
+Refer to
+[this site](https://debois.github.io/elm-mdl/#/menus)
+for a live demo.
+
+# Elm architecture
+@docs Model, model, Action, update, View
+
+# Style
+@docs bottomLeft, bottomRight, topLeft, topRight, unaligned
+-}
+
+import Dict exposing (Dict)
+import Effects exposing (Effects, none)
+import Html.Attributes as Html exposing (..)
+import Html.Events as Html exposing (defaultOptions)
+import Html exposing (..)
+import Json.Decode as Json exposing (Decoder)
+import Json.Encode exposing (string)
+import Material.Helpers exposing (..)
+import String
+import Task
+
+import Material.Menu.Oracle as Oracle exposing (Oracle)
+import Material.Ripple as Ripple
+import Material.Style as Style exposing (Style, cs, cs', css, css', styled)
+
+{-| MDL menu.
+-}
+
+
+-- CONSTANTS
+
+constant :
+  { transitionDurationSeconds  : Float
+  , transitionDurationFraction : Float
+  , closeTimeout               : Float
+  }
+constant =
+  { transitionDurationSeconds  = 0.3
+  , transitionDurationFraction = 0.8
+  , closeTimeout               = 150
+  }
+
+
+-- TODO: Key codes are not implemented yet.
+
+--keycodes =
+--  { enter     = 13
+--  , escape    = 27
+--  , space     = 32
+--  , upArrow   = 38
+--  , downArrow = 40
+--  }
+
+
+-- MODEL
+
+
+{-| Model of the menu; common to all kinds of menus.
+Use `model` to initialise it.
+-}
+
+type alias Model =
+  { alignment : Style
```

Comment:
As far as I can tell, the alignment doesn't need to live in the Model; it is only used in view. Could we take it out of the model and make it a parameter to view instead?

(I was in doubt myself what's the best way to do this for the longest time. I'm convinced by now that in general, you want to take stuff out of the model and put it as argument to view if you can get away with it. The existing components don't always do this yet, but they will eventually.)